kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardHolderUpdateServiceImpl.java

```diff
@@ -0,0 +1,366 @@
+     */
+    private String determineStatusCode (String statusText) {
+        //determine valid status code from status text
+        String statusCode;
+        
+        if (statusText.equals(ACTIVE)) {
+            statusCode = STATUS_CODE_A;
+        } else if (statusText.equals(CARD_ACTIVATION)) {
+            statusCode = STATUS_CODE_T;
+        } else if (statusText.equals(CARD_ACTIVATION_MONITORING)) {
+            statusCode = STATUS_CODE_V;
+        } else if (statusText.equals(MONITORING)) {
+            statusCode = STATUS_CODE_M;
+        } else if (statusText.equals(CARD_HELD)) {
+            statusCode = STATUS_CODE_H;
+        } else if (statusText.equals(CHARGEOFF_ACCOUNT)) {
+            statusCode = STATUS_CODE_E;
+        } else if (statusText.equals(NEW)) {
+            statusCode = STATUS_CODE_N;
+        } else if (statusText.equals(OVERLIMIT)) {
+            statusCode = STATUS_CODE_O;
+        } else if (statusText.equals(PENDING)) {
+            statusCode = STATUS_CODE_P;
+        } else if (statusText.equals(TEMPORARY_LOST_STOLEN)) {
+            statusCode = STATUS_CODE_L;
+        } else if (statusText.equals(TEMPORARILY_BLOCKED_STOLEN)) {
+            statusCode = STATUS_CODE_B;
+        } else {
+            statusCode = STATUS_CODE_C;
+        }
+        return statusCode;
+        
+    }
+    
+    /**
+     * Determines the cancel code from the cancel text
+     */
+    private String determineCancelCode (String statusText) {
+        //determine valid cancel code from status text
+        String cancelCode;
+        
+        if (statusText.equals(CANCELED)) {
+            cancelCode = CANCEL_CODE_E;
+        } else if (statusText.equals(INSTITUTION_SPECIFIC)) {
+            cancelCode = CANCEL_CODE_I;
+        } else if (statusText.equals(TEMPORARY)) {
+            cancelCode = CANCEL_CODE_P;
+        } else if (statusText.equals(PERMANENT)) {
+            cancelCode = CANCEL_CODE_M;
+        } else if (statusText.equals(BANK_REQUEST)) {
+            cancelCode = CANCEL_CODE_B;
+        } else if (statusText.equals(CARDHOLDER_TERMINATED)) {
+            cancelCode = CANCEL_CODE_T;
+        } else if (statusText.equals(CUSTOMER_REQUEST)) {
+            cancelCode = CANCEL_CODE_R;
+        } else if (statusText.equals(COLLECTIONS)) {
+            cancelCode = CANCEL_CODE_C;
+        } else if (statusText.equals(NO_LONGER_NEEDED)) {
+            cancelCode = CANCEL_CODE_N;
+        } else if (statusText.equals(CARDHOLDER_LEFT_COMPANY)) {
+            cancelCode = CANCEL_CODE_A;
+        } else if (statusText.equals(CANCELED_WITHIN_FIVE_DAYS)) {
+            cancelCode = CANCEL_CODE_O;
+        } else if (statusText.equals(LOST_STOLEN)) {
+            cancelCode = CANCEL_CODE_L;
+        } else if (statusText.equals(LOST_STOLEN_FRAUD)) {
+            cancelCode = CANCEL_CODE_D;
+        } else {
+            cancelCode = CANCEL_CODE_X;
+        }
+        return cancelCode;
```

Comment:
Please add a LOG.debug statement at the end of the method to log the resulting statusCode.